src/main/java/ua/hillel/moneymanager/service/expenditure/ExpenditureServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package ua.hillel.moneymanager.service.expenditure;
+
+import jakarta.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import ua.hillel.moneymanager.dto.CategoryDto;
+import ua.hillel.moneymanager.dto.ExpenditureDto;
+import ua.hillel.moneymanager.entity.Expenditure;
+import ua.hillel.moneymanager.mapper.ExpenditureMapper;
+import ua.hillel.moneymanager.repository.CategoryRepository;
+import ua.hillel.moneymanager.repository.ExpenditureRepository;
+import ua.hillel.moneymanager.repository.UserRepository;
+import ua.hillel.moneymanager.service.category.CategoryService;
+import ua.hillel.moneymanager.service.category.CategoryServiceImpl;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Objects;
+
+@Component
+@AllArgsConstructor
+public class ExpenditureServiceImpl implements ExpenditureService {
```

Comment:
A class should be transactional, meaning that you should put the annotation `@Transactional` on the class level. All read methods - should have `@Transactional(readOnly = true)`